dev/checkstyle.xml

```diff
@@ -64,6 +64,11 @@
         <property name="message" value="No trailing whitespace allowed."/>
     </module>
 
+    <module name="RegexpSingleline">
+        <property name="format" value="throw new OutOfMemoryError"/>
+        <property name="message" value="Throw OutOfMemoryError will kill entire executor, replace it with SparkOutOfMemoryError."/>
```

Comment:
Just say that OutOfMemoryError should not generally be thrown by application code. It may not actually be replaceable by SparkOutOfMemoryError

Reply:
Thanks, rewrite this rule in 6e49bb8.